files/grest/cache_queries/epoch_info_cache_setup.sql

```diff
@@ -0,0 +1,204 @@
+DROP TABLE IF EXISTS grest.epoch_info_cache;
+
+DROP TABLE IF EXISTS temp_active_stake;
+
+DROP TABLE IF EXISTS temp_last_block_time;
+
+DROP TABLE IF EXISTS temp_epoch_start_end_time;
+
+CREATE TABLE grest.epoch_info_cache (
+    epoch uinteger PRIMARY KEY NOT NULL,
+    i_out_sum word128type NOT NULL,
+    i_fees lovelace NOT NULL,
+    i_tx_count uinteger NOT NULL,
+    i_blk_count uinteger NOT NULL,
+    i_start_time timestamp without time zone UNIQUE,
+    i_end_time timestamp without time zone UNIQUE,
+    i_first_block_time timestamp without time zone UNIQUE,
+    i_last_block_time timestamp without time zone UNIQUE,
+    i_active_stake lovelace,
+    p_min_fee_a uinteger NOT NULL,
+    p_min_fee_b uinteger NOT NULL,
+    p_max_block_size uinteger NOT NULL,
+    p_max_tx_size uinteger NOT NULL,
+    p_max_bh_size uinteger NOT NULL,
+    p_key_deposit lovelace NOT NULL,
+    p_pool_deposit lovelace NOT NULL,
+    p_max_epoch uinteger NOT NULL,
+    p_optimal_pool_count uinteger NOT NULL,
+    p_influence double precision NOT NULL,
+    p_monetary_expand_rate double precision NOT NULL,
+    p_treasury_growth_rate double precision NOT NULL,
+    p_decentralisation double precision NOT NULL,
+    p_entropy text,
+    p_protocol_major uinteger NOT NULL,
+    p_protocol_minor uinteger NOT NULL,
+    p_min_utxo_value lovelace NOT NULL,
+    p_min_pool_cost lovelace NOT NULL,
+    p_nonce text,
+    p_block_id bigint NOT NULL,
+    p_cost_models character varying,
+    p_price_mem double precision,
+    p_price_step double precision,
+    p_max_tx_ex_mem word64type,
+    p_max_tx_ex_steps word64type,
+    p_max_block_ex_mem word64type,
+    p_max_block_ex_steps word64type,
+    p_max_val_size word64type,
+    p_collateral_percent uinteger,
+    p_max_collateral_inputs uinteger,
+    p_coins_per_utxo_word lovelace
+);
+
+CREATE TEMP TABLE temp_active_stake AS
+SELECT
+    epoch_no,
+    SUM(es.amount) AS i_active_stake
+FROM
+    epoch_stake es
+GROUP BY
+    es.epoch_no;
+
+CREATE TEMP TABLE temp_last_block_time AS SELECT DISTINCT ON (epoch_no)
```

Comment:
Not sure if I understand this one, does this differ greatly from public.epoch table?